addons/pkg-npm/ftests/src/main/java/org/commonjava/indy/pkg/npm/content/NPMHostedReStoreContentTest.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright (C) 2017 Red Hat, Inc. (yma@commonjava.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.indy.pkg.npm.content;
+
+import org.apache.commons.io.IOUtils;
+import org.commonjava.indy.ftest.core.AbstractContentManagementTest;
+import org.commonjava.indy.model.core.HostedRepository;
+import org.commonjava.indy.model.core.StoreKey;
+import org.junit.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+
+import static org.commonjava.indy.pkg.npm.model.NPMPackageTypeDescriptor.NPM_PKG_KEY;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.junit.Assert.assertThat;
+
+/**
+ * This case test if files can be stored in a readonly hosted repo
+ * when: <br />
+ * <ul>
+ *      <li>creates a readonly hosted repo</li>
+ *      <li>stores file in hosted repo once</li>
+ *      <li>updates the hosted repo to non-readonly</li>
+ *      <li>stores file again</li>
+ * </ul>
+ * then: <br />
+ * <ul>
+ *     <li>the file can not be stored with 405 error first time</li>
+ *     <li>the file can be stored successfully with no error second time</li>
+ * </ul>
+ */
+public class NPMHostedReStoreContentTest
```

Comment:
Nope, this is copied from the ReadOnly**Case, will remove the description in next PR.